server.js

```diff
@@ -1,222 +1,25 @@
-}
-if (!cfgConfig.hasOwnProperty('params')) {
-  cfgConfig.params = {};
-}
-if (!cfgConfig.params.hasOwnProperty('defaults')) {
-  cfgConfig.params.defaults = {};
-}
-if (!cfgConfig.params.defaults.hasOwnProperty('baseline')) {
-  cfgConfig.params.defaults.baseline = {};
-}
-if (!cfgConfig.params.defaults.hasOwnProperty('flags')) {
-  cfgConfig.params.defaults.flags = {};
-}
-
-// Combine all variables into one object - ENV is leading!
-const cfgNconf = {
-  env: {
-    ES_PROTOCOL: nconf.get('es_protocol') || cfgConfig.env.ES_PROTOCOL || 'http',
-    ES_HOST: nconf.get('es_host') || cfgConfig.env.ES_HOST || nconf.get('kb_host') ||  cfgConfig.env.KB_HOST || '',
-    ES_PORT: nconf.get('es_port') || cfgConfig.env.ES_PORT || 9200,
-    ES_USER: nconf.get('es_user') || cfgConfig.env.ES_USER || '',
-    ES_PASS: nconf.get('es_pass') || cfgConfig.env.ES_PASS || '',
-    ES_SSL_CERT: nconf.get('es_ssl_cert') || cfgConfig.env.ES_SSL_CERT || '',
-    ES_SSL_KEY: nconf.get('es_ssl_key') || cfgConfig.env.ES_SSL_KEY || '',
-    KB_HOST: nconf.get('kb_host') || cfgConfig.env.KB_HOST || nconf.get('es_host') ||  cfgConfig.env.ES_HOST || '',
-    KB_PORT: nconf.get('kb_port') || cfgConfig.env.KB_PORT || 5601,
-    KB_INDEX: nconf.get('kb_index') || cfgConfig.env.KB_INDEX || '.kibana',
-    KB_RENAME: nconf.get('kb_rename') || cfgConfig.env.KB_RENAME || '',
-    HTTP_PORT: nconf.get('http_port') || cfgConfig.env.HTTP_PORT || 80,
-    DEBUG: nconf.get('debug') || cfgConfig.env.DEBUG || false,
-    APP_NAME: pkg.name,
-    APP_VERSION: pkg.version,
-    APP_CONFIG: cfgFile,
-    HOST: os.hostname(),
-    NODE_ENV: nconf.get('node_env') || 'development',
-    INDEX_PERF: 'cicd-perf',
-    INDEX_RES: 'cicd-resource',
-    INDEX_ERR: 'cicd-errorlog',
-    useES: false
-  },
-  params: {
-    required: cfgConfig.params.required || ['log.test_info', 'log.env_tester', 'log.team', 'log.browser', 'log.env_target'],
-    defaults: {
-      baseline: {
-        days: cfgConfig.params.defaults.baseline.days || 7,
-        perc: cfgConfig.params.defaults.baseline.perc || 75,
-        padding: cfgConfig.params.defaults.baseline.padding || 1.2
-      },
-      flags: {
-        assertBaseline: cfgConfig.params.defaults.flags.assertBaseline || true,
-        debug: cfgConfig.params.defaults.flags.debug || false,
-        esTrace: cfgConfig.params.defaults.flags.esTrace || false,
-        esCreate: cfgConfig.params.defaults.flags.esCreate || false,
-        passOnFailedAssert: cfgConfig.params.defaults.flags.passOnFailedAssert || false
-      }
-    }
-  }
-};
-
-// Load config object into nconf
-nconf
-  .use('memory')
-  .defaults(cfgNconf);
-
+const app = require('./src/app');
 const env = nconf.get('env');
 
-if (env.DEBUG !== true) {
-  logger.transports.console.silent = true;
+// Check ES (if in use)
+if (env.ES_HOST && typeof env.ES_HOST === 'string') {
+  const runES = require('./src/run-es');
+  const checkES = new runES.Elastic();
+  checkES.setup()
```

Comment:
shouldn't we add an error handler?

Reply:
Error handling is done in `'./src/run-es'` ... if there is a communication issue with elasticsearch, `run-es` will log the error end move on - the server should still start listening (i.e. call `startListen()`) ... is that a good way to do things?